### DISCUSSION TITLE
How to improve tracking and counting accuracy?

### DISCUSSION THREAD

**Discussion by rafaelgildin** [roboflow/supervision#785] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Based on the great video from Piotr [here](https://www.youtube.com/watch?v=OS5qI9YBkfk) I generated [this code](https://github.com/rafaelgildin/track_and_count_people/blob/master/how-to-track-and-count-people-with-yolov8.ipynb) which produced the [output](https://github.com/rafaelgildin/track_and_count_people/blob/master/data/public_walk_output.mp4).
Based on the output it's possible to see some miscounts from the model and here's a list of the possible hypothesis followed by it's questions:

1. Some people don't show up with it's entire body before and after crossing the line. How to improve tracking and counting accuracy? Here I mean how to improve the model counts, consider counting other parts of the body, not just the entire body. For example, can I train the model with a dataset from heads as being count as people? Is that going to improve the acc?

2. How can I discover if it's a detection or tracking problem. Based on that, training the model with a custom dataset like described in the first topic, if it's a detection problem; or if it's a tracking one, changing the tracker or tunning it's parameters is going to improve it's acc?

Based on them, I greatly appreciate if you can answer with your thoughts and I'll be happy for any advice too.
Finally, I declared it as question not a bug or future request, because it envolves ultralytics library which could be the current problem too.

**Comment by SkalskiP** [comment]
First, let me convert it from an issue to a discussion and put it into the Q&A section.

**Comment by SkalskiP** [comment]
1. The code we show in this video is quite old - supervision-0.1.0 if I'm not mistaken. Since then, a lot has changed. One of the community members has made a refreshed version of the notebook using new supervision features. Use [this](https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-track-and-count-vehicles-with-yolov8-and-supervison.ipynb) code as a reference point.
2. Next time, also mark the detections with their tracker ID and box. It will make it much easier for us to debug.
3. By default, sv.LineZone uses the four corners of the bounding box as the triggering criterion. This means that all four corners of the box must first be on one side of the line and all four must cross it.
4. Consider running the model at a higher resolution. Instead of `model = YOLO(MODEL)`, use `model = YOLO(MODEL, imgsz=1280)` - this will allow you to detect more objects.

**Reply by rafaelgildin** [reply]
@SkalskiP thanks a lot for the answers. Answering the topics, [code updated](https://github.com/rafaelgildin/track_and_count_people/tree/master)

1. I'm already using it. Supervision version '0.17.0'
2. Tracker id is on. I changed change to box.
3.Got it, but by training with just the heads would it improve the counts?
4. Made the change, but the result is the same (in:7, out:8)

Any other advice please?
